services/provider/server/server.go

```diff
@@ -325,6 +325,11 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 				return nil, err
 			}
 
+			cephClientAnnotation, err := s.getCephClientAnnotation(ctx, i.Name)
+			if err != nil {
+				return nil, err
+			}
+
```

Comment:
Why are we reading the resource twice? (once in `getCephClientSecretName` and immediatly after in `getCephClientAnnotation`?

I would suggest to refactor  `getCephClientSecretName` into a `getCephClientInformation` util which return what you need:

```
func (s *OCSProviderServer) getCephClientInformation context.Context, name string) (string, string, error) 
```

where the first return value is a string for the name and the second is a bool for cephUserType